File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/SettingsAct/settingAct.java
package com.example.adminapp.SettingsAct;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Toast;

import com.example.adminapp.MainActivity;
import com.example.adminapp.R;
import com.example.adminapp.databinding.ActivitySettingBinding;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class settingAct extends AppCompatActivity {

    private ActivitySettingBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySettingBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        actClicked();
    }

    private void actClicked() {
        binding.profileSettingTvSettingsAct.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(getApplicationContext(),settingFrManagerAct.class);
                i.putExtra("view",getTag(view));
//                Log.i("Id ====== ",getTag(view));
                startActivity(i);
            }
        });
        binding.tNcTvSettingsAct.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(getApplicationContext(),settingFrManagerAct.class);
                i.putExtra("view",getTag(view));
//                Log.i("Id ====== ",getTag(view));
                startActivity(i);
            }
        });
        binding.signOutTvSettingsAct.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(getApplicationContext(),settingFrManagerAct.class);
                i.putExtra("view",getTag(view));
//                Log.i("Id ====== ",getTag(view));
                startActivity(i);
            }
        });

    }

    private String getTag(View view){
        String tag = (String) view.getTag();
        return tag;
    }
}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/SettingsAct/settingAct.kt
package com.example.adminapp.SettingsAct

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.adminapp.databinding.ActivitySettingBinding

class settingAct : AppCompatActivity() {
    private var binding: ActivitySettingBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingBinding.inflate(layoutInflater)
        setContentView(binding!!.root)
        actClicked()
    }

    private fun actClicked() {
        binding!!.profileSettingTvSettingsAct.setOnClickListener { view ->
            val i = Intent(applicationContext, settingFrManagerAct::class.java)
            i.putExtra("view", getTag(view))
            //                Log.i("Id ====== ",getTag(view));
            startActivity(i)
        }
        binding!!.tNcTvSettingsAct.setOnClickListener { view ->
            val i = Intent(applicationContext, settingFrManagerAct::class.java)
            i.putExtra("view", getTag(view))
            //                Log.i("Id ====== ",getTag(view));
            startActivity(i)
        }
        binding!!.signOutTvSettingsAct.setOnClickListener { view ->
            val i = Intent(applicationContext, settingFrManagerAct::class.java)
            i.putExtra("view", getTag(view))
            //                Log.i("Id ====== ",getTag(view));
            startActivity(i)
        }
    }

    private fun getTag(view: View): String {
        return view.tag as String
    }
}